infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -96,9 +89,10 @@ resource functionAppSettings 'Microsoft.Web/sites/config@2023-01-01' = {
     WEBSITE_CONTENTAZUREFILECONNECTIONSTRING: storageAccountConnectionString
     WEBSITE_CONTENTSHARE: toLower(functionAppName)
     FUNCTIONS_EXTENSION_VERSION: '~4'
-    APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsightsModule.outputs.applicationInsightsKey
+    APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsightsKey
     FUNCTIONS_WORKER_RUNTIME: functionAppRuntime
     WEBSITE_RUN_FROM_PACKAGE: '1'
+    virtualNetworkSubnetId: subnetId
```

Comment:
As a part of sorting this out, it highlighted that we need to move to a Function App hosting plan that supports VNet integration, or otherwise abandon the VNet integration altogether.  For now I have bumped this onto Premium just within the Function App Bicep template itself, but moving forward we would probably change this to:
- Have the `functionApp.bicep` module accept a SKU as a parameter, and use a strongly-typed param or user-defined type to describe all the available SKU structures / options
- Have each environment's `bicepparam` file provide the appropriate SKU, potentially with a sensible default in `main.bicep`.